File: decompiled/sources/com/hjq/http/body/StringBody.java
package com.hjq.http.body;

import com.hjq.http.model.ContentType;
import java.io.IOException;
import okhttp3.MediaType;
import okhttp3.RequestBody;
import okio.BufferedSink;

public class StringBody extends RequestBody {
    private final byte[] mBytes;
    private final String mText;

    public StringBody() {
        this("");
    }

    public StringBody(String str) {
        this.mText = str;
        this.mBytes = str.getBytes();
    }

    public MediaType contentType() {
        return ContentType.TEXT;
    }

    public long contentLength() {
        return (long) this.mBytes.length;
    }

    public void writeTo(BufferedSink bufferedSink) throws IOException {
        byte[] bArr = this.mBytes;
        bufferedSink.write(bArr, 0, bArr.length);
    }

    public String toString() {
        return this.mText;
    }
}

File: decompiled/sources/com/hjq/http/body/StringBody.kt
package com.hjq.http.body

import com.hjq.http.model.ContentType
import java.io.IOException
import okhttp3.MediaType
import okhttp3.RequestBody
import okio.BufferedSink

open class StringBody(
    private val mText: String = ""
) : RequestBody() {
    private val mBytes: ByteArray = mText.toByteArray()

    override fun contentType(): MediaType {
        return ContentType.TEXT
    }

    override fun contentLength(): Long {
        return mBytes.size.toLong()
    }

    @Throws(IOException::class)
    override fun writeTo(bufferedSink: BufferedSink) {
        bufferedSink.write(mBytes, 0, mBytes.size)
    }

    override fun toString(): String {
        return mText
    }
}
